client/verta/tests/unit_tests/test_runtime.py

```diff
@@ -202,4 +202,14 @@ def test_s3_validation_good_keys(key: str):
     runtime._validate_s3(key)
 
 
-
+def test_overwrite_existing_key_throws_error() -> None:
+    """
+    If an attempt is made to write to the same key twice within a single
+    context, a KeyError exception is raised.
+    """
+    with pytest.raises(KeyError)as err:
+        with runtime.context() as ctx:
+            runtime.log('key_123', 'value_1')
+            runtime.log('key_123', 'value_2')
```

Comment:
As I've mentioned in a few comments before already, I'd encourage making the `with pytest.raises(...):` body _as specific as possible_ in what it contains.

```suggestion
    with runtime.context() as ctx:
        runtime.log('key_123', 'value_1')
        with pytest.raises(KeyError)as err:
            runtime.log('key_123', 'value_2')
```

Reply:
Done.